main.py

```diff
@@ -41,11 +58,19 @@ def index():
 
     pubsub_message = envelope['message']
 
-    name = 'World'
+    response = 'empty response'
     if isinstance(pubsub_message, dict) and 'data' in pubsub_message:
```

Comment:
I think this could be more clear as:

if not isinstance(pubsub_message, dict)
   handle error
if 'data' not in pubsub_message:
  handle error

monitoring_notification_string = base64.b64decode(pubsub_message['data']).decode('utf-8').strip()
try:
monitoring_notification_dict = json.loads(monitoring_notification_string)
except json.JSONDecodeError:

Reply:
So I've been looking into ways to validate messages and I think using the jsonschema library would make this a lot cleaner. Here's an example of how it is used: https://medium.com/grammofy/testing-your-python-api-app-with-json-schema-52677fe73351